src/main/java/org/opensearch/security/securityconf/ConfigModelV7.java

```diff
@@ -930,6 +931,15 @@ private static String replaceRoles(final String orig, final User user) {
         return retVal;
     }
 
+    private static String replaceSecurityRoles(final String orig, final User user) {
+        String retVal = orig;
+        if (orig.contains("${user.securityRoles}") || orig.contains("${user_securityRoles}")) {
```

Comment:
What is `user_securityRoles`

Reply:
I included the `user_securityRoles` as the underscore seemed to be supported for the other two parameter substitutions for `user_name` and `user_roles`. I have removed now.

Reply:
I didn't mean that `user_securityRoles` shouldn't be removed. I actually have a question on this. Is it needed here? Can we make sure that it exists in DLS query and index pattern, and it should be supported?

Reply:
I can add it back in and would you like me to add a test for it or is there something else I also need to add?

Reply:
Sorry for the confusion if the question wasn't clear! I see the following in the tests. `securityRoles` are being populated in the requests. Is `user_securityRoles` another field supported in requests? Or is it an alternative of `securityRoles` in requests? Could you please help me understand this?
```
       tc.index(new IndexRequest("prop-mapped").type("_doc").setRefreshPolicy(RefreshPolicy.IMMEDIATE)
            .source("{\"securityRole\": \"opendistro_security_mapped\", \"amount\": 6060}", XContentType.JSON)).actionGet();
        tc.index(new IndexRequest("prop-mapped").type("_doc").setRefreshPolicy(RefreshPolicy.IMMEDIATE)
            .source("{\"securityRole\": \"not_assigned\", \"amount\": 7070}", XContentType.JSON)).actionGet();
```

Reply:
I'm only testing the `user.securityRoles`in my test with the following DLS query
```
    dls: "{\"terms\" : { \"securityRole\" : [${user.securityRoles}]}}"
```
I then add two files to my `prop-mapped` index, each requiring a different security role. I assigned the backend role of `prolemapped` to the user and this maps to the security role `opendistro_security_mapped`. So the user is able to read only the one document that they have the role for. If I wanted to use the `user_securityRoles` substitution then I could modify the DLS query to the following:
```
dls: "{\"terms\" : { \"securityRole\" : [${user_securityRoles}]}}"
```
If I add the underscore substitution option back in then this would have the same outcome for the user.

My guess is that the underscore substitution was to maintain backwards compatibility with older versions as it is an undocumented feature.

So, I can add it back in if we want to support both options?

Reply:
Ok. Please add it back then. Thanks!

Reply:
Have added it back in. Thanks @cliu123 and @peternied for looking at this for us :)